x-pack/metricbeat/module/containerd/memory/memory.go

```diff
@@ -0,0 +1,161 @@
+			"container_memory_swap_limit_bytes":          prometheus.Metric("swap.limit"),
+			"container_memory_swap_failcnt_total":        prometheus.Metric("swap.fail.count"),
+			"container_memory_total_inactive_file_bytes": prometheus.Metric("inactiveFiles"),
+			"container_memory_total_active_file_bytes":   prometheus.Metric("activeFiles"),
+			"container_memory_total_cache_bytes":         prometheus.Metric("cache"),
+			"container_memory_total_rss_bytes":           prometheus.Metric("rss"),
+		},
+		Labels: map[string]prometheus.LabelMap{
+			"container_id": prometheus.KeyLabel("id"),
+			"namespace":    prometheus.KeyLabel("namespace"),
+		},
+	}
+)
+
+// Metricset for containerd memory is a prometheus based metricset
+type metricset struct {
+	mb.BaseMetricSet
+	prometheusClient prometheus.Prometheus
+	mod              containerd.Module
+	calcPct          bool
+}
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	mb.Registry.MustAddMetricSet("containerd", "memory", New,
+		mb.WithHostParser(hostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The containerd cpu metricset is beta.")
+
+	pc, err := prometheus.NewPrometheusClient(base)
+	if err != nil {
+		return nil, err
+	}
+	config := containerd.DefaultConfig()
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	mod, ok := base.Module().(containerd.Module)
+	if !ok {
+		return nil, fmt.Errorf("must be child of kubernetes module")
+	}
+	return &metricset{
+		BaseMetricSet:    base,
+		prometheusClient: pc,
+		mod:              mod,
+		calcPct:          config.CalculateMemPct,
+	}, nil
+}
+
+// Fetch gathers information from the containerd and reports events with this information.
+func (m *metricset) Fetch(reporter mb.ReporterV2) error {
+	families, _, err := m.mod.GetContainerdMetricsFamilies(m.prometheusClient)
+	if err != nil {
+		return errors.Wrap(err, "error getting families")
+	}
+	events, err := m.prometheusClient.ProcessMetrics(families, mapping)
+	if err != nil {
+		return errors.Wrap(err, "error getting events")
+	}
+
+	for _, event := range events {
+
+		// setting ECS container.id and module field containerd.namespace
+		containerFields := common.MapStr{}
+		moduleFields := common.MapStr{}
+		rootFields := common.MapStr{}
+
+		cID := containerd.GetAndDeleteCid(event)
+		namespace := containerd.GetAndDeleteNamespace(event)
+
+		containerFields.Put("id", cID)
+		rootFields.Put("container", containerFields)
+		moduleFields.Put("namespace", namespace)
+
+		// Calculate memory total usage percentage
+		if m.calcPct {
+			inactiveFiles, err := event.GetValue("inactiveFiles")
+			if err != nil {
+				continue
```

Comment:
in which cases `inactiveFiles` can be not present in event? should here be added a debug log to explain why usage percentage was skipped?

Reply:
In no case. Only if there is an error. I will add a debug